kernel/src/path.rs

```diff
@@ -176,19 +182,26 @@ impl<Location: AsUrl> ParsedLogPath<Location> {
     }
 }
 
+impl ParsedLogPath<Url> {
+    /// Create a new ParsedCommitPath<Url> for a new json commit file at the specified version
+    pub(crate) fn new_commit(
+        table_root: &Url,
+        version: Version,
+    ) -> DeltaResult<ParsedLogPath<Url>> {
+        let filename = format!("{:020}.json", version);
+        let location = table_root.join("_delta_log/")?.join(&filename)?;
+        let path = Self::try_from(location)?.expect("valid commit path");
+        assert!(path.is_commit());
+        Ok(path)
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use std::path::PathBuf;
 
     use super::*;
 
-    // Easier to test directly with Url instead of FileMeta!
```

Comment:
so that i could `impl ParsedLogPath<Url> { ... }` for the `new_commit` - do you think it's useful to try to make that generic over `Location` too? I think that would require some more involved changes in order to let us actually create a `Location`? I went back and forth on whether or not my use case fit the `ParsedLogPath` pattern but ultimately decided to use is just for clarity/safety

Reply:
That makes sense!